### PR TITLE
Revised home page layout

### PR DESCRIPTION
**Pull request by ebeneliason**
This tweaks copy and layout on the homepage to provide a better
visual hierarchy. It also introduces some changes to the Card
component and styles to support the new look in cleaner fashion.

### CODE REVIEW

**Comment by josh-padnick** [comment]
I was just browsing this and this looks awesome! Some additional small feedback:

- **What is Gruntwork?** Shorten text to "<div>Learn how Gruntwork products can help you deploy a world class infrastructure.</div>"
- I love the RefArch vs "Deploy a Service" breakdown. The icons are great, too!
- **Setup your accounts.** Should this be "Setup Your AWS Multi-Account Structure"?
- **Create a CI/CD Pipeline.** Adjust to "Create a CI/CD Pipeline for Infra." Might be too long, but we want to clarify it's for infra.
- Change "deploy a k8s cluster" to "Deploy Kubernetes using EKS"?
- Finally, perhaps there's some kind of common icon we can use to signify a "use case"? Maybe a single cube, or a document icon or something else. 

These are all off-the-cuff ideas. You should make the final call on everything here.

File: src/components/Grid.tsx
Comment (eak12913) [inline src/components/Grid.tsx:37]: This is more semantically correct right? `section` refers more to a "section" in a document rather than this..being a grid, which is more of a container. Am I understanding this properly?

File: src/pages/index.tsx
Comment (eak12913) [inline src/pages/index.tsx:52]: Nit: This feels more like a marketing blurb and less like verbiage that will help customers find docs for interacting with their Ref Arch.

I could see something like: "Got a Ref Arch deployed? Learn all about it" <-- not this exact text...but the gist should be "if you got the ref arch, this is where you can learn about how to absorb it into your org"

**Review by eak12913 — APPROVED** [review]
I had a minor nit - but it's not a blocker. So I'm approving this and can re-stamp if we do end up making changes.

File: src/components/Grid.tsx
Comment (ebeneliason) [reply src/components/Grid.tsx:37]: I had the same thought initially, but in practice I noticed a couple of related things that changed my mind.
1. In some cases the Grid belongs logically grouped with some other content, e.g. a header, or supplemental text. In those cases, the semantic "section" is all of that content taken together, not _just_ the grid.
2. In some cases, the component using the grid already had a higher level notion of _section_, which resulted in nested sections. Nested sections aren't prohibited, but semantically it was again incorrect, since the section is meant to map to a section of the page that would be shown in a document outline.

File: src/pages/index.tsx
Comment (ebeneliason) [reply src/pages/index.tsx]: Yeah, that's fair. Updated!

**Review by eak12913 — APPROVED** [review]
LGTM!

**Pull request merged**
